app/src/main/java/com/cmput301f20t21/bookfriends/ui/login/LoginActivity.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * LoginActivity.java
+ * Version: 1.0
+ * Date: October 18, 2020
+ * Copyright (c) 2020. Book Friends Team
+ * All rights reserved.
+ * github URL: https://github.com/CMPUT301F20T21/Book_Friends
+ */
+
+package com.cmput301f20t21.bookfriends.ui.login;
+
+import android.os.Bundle;
+import android.text.Editable;
+import android.view.View;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.cmput301f20t21.bookfriends.R;
+import com.google.android.material.textfield.TextInputEditText;
+import com.google.android.material.textfield.TextInputLayout;
+
+/**
+ * Activity that will display the login page and handle user inputs
+ */
+public class LoginActivity extends AppCompatActivity {
+    private TextInputLayout usernameLayout;
+    private TextInputLayout passwordLayout;
+    private TextInputEditText usernameField;
+    private TextInputEditText passwordField;
+
+    /**
+     * android lifecycle method, grab the Layout and EditText fields
+     * as well as adding a simple textChangedListener
+     * @param savedInstanceState - the saved objects, should contain nothing for this activity
+     */
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_login);
+        usernameLayout = findViewById(R.id.login_username_layout);
+        usernameField = findViewById(R.id.login_username_field);
+        usernameField.addTextChangedListener(new AfterTextChangedWatcher() {
+            @Override
+            public void afterTextChanged(Editable usernameString) {
+                String username = usernameString.toString();
+                checkUsername(username); // return value not needed
```

Comment:
This will set an error when user type 1 character and delete. I don't think that's necessary. We can set error only once when login is clicked. We could make afterTextChanged setError to null only when user types.

Reply:
fixed